src/pegin-address-verifier.js

```diff
@@ -1,19 +1,71 @@
-var p2pkhP2sh = require('./crypto/p2pkh-p2sha');
-var segwit = require('./crypto/segwit-addr');
-let { ADDRESS_TYPES } = require('./crypto/constants');
+const p2pkhP2sh = require('./crypto/p2pkh-p2sha');
+const segwit = require('./crypto/segwit-addr');
+const { ADDRESS_TYPES, ADDRESS_TYPES_CODES } = require('./crypto/constants');
 
-module.exports = {
-    isValidAddress: function (address, networkType) {
-        return p2pkhP2sh.isValid(address, networkType) || 
-        segwit.isValidAddress(address, networkType);
-    },
-    getAddressInformation: (address) => {
-        return p2pkhP2sh.getAddressInfo(address) || 
-        segwit.getAddressInfo(address);
-    },
-    canPegIn: (addressInfo) => {
-        return addressInfo && 
-            addressInfo.type == ADDRESS_TYPES.P2PKH || 
-            addressInfo.type == ADDRESS_TYPES.P2SH;
+const RSKT_PREFIX_HEX = '52534b54'; // 'RSKT' prefix encoded in hex. Check https://github.com/rsksmart/RSKIPs/blob/2c994cc108885ccc5a116e4aee8c073b5eca5682/IPs/RSKIP170.md
+const RSKT_PROTOCOL_VERSION = '01';
+
+const isValidAddress = (address, networkType) => {
+    return p2pkhP2sh.isValid(address, networkType) || 
+    segwit.isValidAddress(address, networkType);
+};
+
+const getAddressInformation = (address) => {
+    return p2pkhP2sh.getAddressInfo(address) || 
+    segwit.getAddressInfo(address);
+};
+
+const canPegIn = (addressInfo) => {
+    return addressInfo && 
+        addressInfo.type == ADDRESS_TYPES.P2PKH || 
+        addressInfo.type == ADDRESS_TYPES.P2SH;
+};
+
+/**
+ * Creates the pegin v1 data for the given {rskDestinationAddress} and optionally the {btcRefundAddress}.
+ * @param {string} rskDestinationAddress 
+ * @param {string?} btcRefundAddress 
+ * @returns {string} The pegin v1 data.
+ */
+const createPeginV1TxData = (rskDestinationAddress, btcRefundAddress) => {
+
+    if(!rskDestinationAddress) {
+        throw new Error('RSK destination address is required');
     }
+
+    let data = `${RSKT_PREFIX_HEX}${RSKT_PROTOCOL_VERSION}`;
+    
+    if (rskDestinationAddress.startsWith('0x')) {
+        rskDestinationAddress = rskDestinationAddress.substring(2);
+    }
+
+    data += rskDestinationAddress;
+
+    if (btcRefundAddress) {
+        const refundAddressInfo = getAddressInformation(btcRefundAddress);
+        if (refundAddressInfo) {
+            data += ADDRESS_TYPES_CODES[refundAddressInfo.type];
```

Comment:
What happened if the type is not found in the ADDRESS_TYPES_CODES?

I prefer to move this below to the case code block where you know what type is the refund address.

Reply:
It would throw an exception.